### DISCUSSION TITLE
Mapping of heating profiles between HomematicIP and Home Assistant?

### DISCUSSION THREAD

**Discussion by mzurhorst** [SukramJ/aiohomematic#793] [Q&A] [answered]
Hi all,

I just recently started with HomematicIP in Home Assistant, and I am a bit confused how the profiles are mapped:

1. When I look into into Raspberrymatic CCU for an eTRV, I see "week programs"  1-3.
2. When creating a "heating group", I now have "week profiles" 1-6.
3. And Home Assistant provides "preset_modes"  (boost, none + Profile 1-3)

I was looking for a comprehensive documentation of how those settings are related, but I failed.
Which programs/profiles to I have to configure in Raspberrymatic so that I can toggle them from Home Assistant with my automations?

Thanks and best regards,
   Marcus

**Home Assistant:**
![grafik](https://user-images.githubusercontent.com/575190/210048431-fbc86785-8a70-4365-8961-6c315aae1846.png)

**HomematicIP  eTRV:**
![grafik](https://user-images.githubusercontent.com/575190/210047715-62a25f83-20b8-4913-b096-e2dd27df8d1a.png)

**HomematicIP Heizgruppe:**
![grafik](https://user-images.githubusercontent.com/575190/210048531-b04387b5-9505-411e-b048-970dd479cfb1.png)

**Comment by SukramJ** [comment, marked as answer]
Then number of available week_programs is dynamically read out from the running device/group in HA.

TRVs have 3 week_programs
Wallthermostats (like BWTH) have 6 week_programs

If you put only a TRV in a group this group will have 3 week_programs.
If you put only a Wallthermostat in a group this group will have 6 week_programs.
If you put a TRV and a Wallthermostat in a group this group will have 6 week_programs.

The CCUs WebUI always displays 6 week_programs.

So if you use week_program 4 in a group with only a TRV then this cannot be applied to the TRV.

**Reply by mzurhorst** [reply]
Thank you. 👍 

So I would be safe when I stick to week_programs 1-3 only, and those match the "profile_{1..3}" in Home Assistant, correct?
--- That would mean that I have to reconfigure in Raspberrymatic.      

Btw, is it required (or recommended) to have TRV and heat group settings identical in RM? -- Or does the heat group have priority and ignore the settings of the individual devices?

**Reply by SukramJ** [reply]
> So I would be safe when I stick to week_programs 1-3 only, and those match the "profile_{1..3}" in Home Assistant, correct?

Yes

> Btw, is it required (or recommended) to have TRV and heat group settings identical in RM? -- Or does the heat group have priority and ignore the settings of the individual devices?

Groups superseed settings of devices. For more information see official eQ-3 docs or check the [homematic forum](https://homematic-forum.de/).

**Comment by kwithus** [comment]
Hi, with the newest update the week-programs show up as week-program in HA, no longer as ‚preset‘, which is nice.
However, is there a way to label those individually, either in HM or HA? E.g. ‚Normal‘  instead of week_programm_1 ?

**Reply by SukramJ** [reply]
???
They are still presets but renamed from `Profile 1` to `week_program_1`.

> However, is there a way to label those individually

No, but these presets are translated to `WP 1` in the UI

**Comment by kwithus** [comment]
Yes, of course renamed from profile to Week Programm,
However,  I don’t see WP-1 but Week_programm_1
![D3FE480F-78F5-487D-9BD0-5709C243167C](https://user-images.githubusercontent.com/20025320/210747715-009b0e31-d340-4915-bd38-a0cef1b985fc.png)

**Reply by SukramJ** [reply]
Which language (code) do you use?
Maybe you need to refresh your browser cache.

**Comment by kwithus** [comment]
Thanks, deleting the cache for this website in Safari and refreshing the HASS App did it. (German language btw)

**Comment by fmiermont** [comment]
Hello,

I think there is something I do not understand clearly. When a Group have a thermostat, we have access to 6 week program on the group settings.
But I do not see them on the group device in HA ? The thermostat is ok, but not the group.

Did I miss something here ?

Thanks !

**Reply by SukramJ** [reply]
> Did I miss something here ?

Screenshots?

**Reply by fmiermont** [reply]
so I guess there is something wrong :)
Here are some screens :)
group in raspberrymatic with 6 profiles :
![rasphmipgroup](https://user-images.githubusercontent.com/68554923/210788148-a335d85d-d98f-4471-a5e1-0643531dcd7b.png)

Now the same in HA
<img width="428" alt="hmipGroup" src="https://user-images.githubusercontent.com/68554923/210788237-cc3564ae-ab90-46e1-8a42-05fed5a7618d.png">

the Thermostat is ok though 
<img width="435" alt="hmipThm" src="https://user-images.githubusercontent.com/68554923/210788282-8fc9e702-ae3e-4744-8727-91d75e3c3596.png">


Regards,

**Reply by SukramJ** [reply]
Maybe I am wrong and groups always have only 3 week_program, and **i don't care what the UI displays**!

This is not a configuration option. We use the parameter ACTIVE_PROFILE to get the active week program.
This parameter also has a min and a max value, that we use to determine the number of available week programs.

You should use the service `Homematic(IP) Local: Clear the cache`, restart HA and see what happens.

**Reply by SukramJ** [reply]
You should also refresh your browser cache `week_program_1 `is the raw state value and should be displayed as `WP 1`

**Reply by fmiermont** [reply]
After clearing the cache (service and browser), now I see WP indeed, but only 1/2/3.

Not sure to understand what you means when you say you don't care what the UI displays :) Do you mean that it's rapberrymatic that is bugged and should not display 6 WP on a group even if it contains a thermostat ?

**Reply by SukramJ** [reply]
> Not sure to understand what you means when you say you don't care what the UI displays 
Here i mean the CCUs WebUI, that displays sometimes things different from what we see/can use from the API.

You could check with the RaspberryMatic UI what profiles should be available:
- System Control -> DevConfig -> Device List -> (search your device) 
- On channel HEATING_CLIMATECONTROL_TRANSCEIVER (xxxx:1) scroll down to the VALUES section:
- ACTIVE_PROFILE should display the current value and at the end the min/max.

Repeat the same for the group.

What devices do you have in the Group?

**Reply by fmiermont** [reply]
Hum I'm really sorry I don't know where to find any "DevConfig" under rasperrymatic ? I'm a noob with HA (but i've already enable the expert mode) it's in version 3.65.11.20221005 :/

**Reply by SukramJ** [reply]
Maybe you need to enable it:
- System Control -> Advanced Settings
- Scroll down

**Reply by fmiermont** [reply]
Indeed it was not activated, so now I can check the ACTIVE_PROFILE and you're right, even if my thermostat has the 6 values, the group doesn't.
That's weird but the issue is RaspberryMatic related, not hahomeatic :)

Thanks a lot @SukramJ :)

**Reply by Baxxy13** [reply]
It is an decision from eQ-3, not from RaspberryMatic. 
IP-Groups have 6 Profiles, you can select them via the ui.

But in the Metadata the are limited to 3.
So everything that reads the Metadata is limited to 3, also WebUI-Programms.
